test/message/connection-endpointSpec.js

```diff
@@ -227,7 +227,7 @@ describe( 'connection endpoint', function() {
 
 			socketMock.emit( 'message', _msg( 'A|REQ|{"user":"wolfram"}+' ) );
 
-			expect( socketMock.lastSendMessage ).toBe( _msg( 'A|A+' ) );
```

Comment:
Would be good to keep the previous test and create a new one that returns both the user and data to ensure both scenarios work.

In the PermissionHandlerMock you can add a condition to return the params only when needed. 

To stick with the approach used, you would add a 

`permissionHandlerMock.nextUserValidationReturnData = 'test-data'`

and that would allow you to return the data you assigned, or undefined by default.

```
if( this.nextUserValidationResult === true ) {
    callback( null, 'test-user', nextUserValidationReturnData  );
 }
```